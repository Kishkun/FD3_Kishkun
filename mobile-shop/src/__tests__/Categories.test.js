import {mount} from "enzyme";
import React from "react";
import ConnectedCategories from "../components/categories/Categories";
import {Provider} from "react-redux";
import configureMockStore from "redux-mock-store";
import { BrowserRouter as Router } from 'react-router-dom';

test("testing BasketCart component", () => {

    let categories = [
        {
            "id": "1",
            "name": "Apple"
        }
    ];

    const initialState = {
        categories: categories
    };
    const mockStore = configureMockStore();
    let store, wrapper;

    store = mockStore(initialState);
    wrapper = mount(<Provider store={store}>
        <Router>
            <ConnectedCategories/>
        </Router>
    </Provider>);

    expect(wrapper).toMatchSnapshot();
});