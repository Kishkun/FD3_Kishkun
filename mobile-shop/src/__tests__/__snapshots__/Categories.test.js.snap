// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing BasketCart component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(Connect(Categories))>
        <Connect(Categories)
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <Categories
            categories={
              Array [
                Object {
                  "id": "1",
                  "name": "Apple",
                },
              ]
            }
            dispatch={[Function]}
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <div
              className="well"
            >
              <h4>
                Brand
              </h4>
              <div
                className="list-group"
              >
                <Link
                  className="list-group-item active"
                  to="/"
                >
                  <LinkAnchor
                    className="list-group-item active"
                    href="/"
                    navigate={[Function]}
                  >
                    <a
                      className="list-group-item active"
                      href="/"
                      onClick={[Function]}
                    >
                      All
                    </a>
                  </LinkAnchor>
                </Link>
                <Link
                  className="list-group-item"
                  key="1"
                  to="/categories/1"
                >
                  <LinkAnchor
                    className="list-group-item"
                    href="/categories/1"
                    navigate={[Function]}
                  >
                    <a
                      className="list-group-item"
                      href="/categories/1"
                      onClick={[Function]}
                    >
                      Apple
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </div>
          </Categories>
        </Connect(Categories)>
      </withRouter(Connect(Categories))>
    </Router>
  </BrowserRouter>
</Provider>
`;
